qa/rpc-tests/mempool_tx_expiry.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python2
+# Copyright (c) 2018 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#
+# Test proper expiry for transactions >= version 3
+#
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.authproxy import JSONRPCException
+from test_framework.util import assert_equal, connect_nodes, \
+    connect_nodes_bi, sync_blocks, start_node
+
+class MempoolTxExpiryTest(BitcoinTestFramework):
+
+    def setup_network(self, split=False):
+        self.nodes = []
+        # Start nodes with tiny block size of 11kb
+        self.nodes.append(start_node(0, self.options.tmpdir, ["-txblockexpiry=2", "-blockprioritysize=7000", "-blockmaxsize=11000", "-maxorphantx=1000", "-relaypriority=true", "-printpriority=1"]))
+        self.nodes.append(start_node(1, self.options.tmpdir, ["-txblockexpiry=2", "-blockprioritysize=7000", "-blockmaxsize=11000", "-maxorphantx=1000", "-relaypriority=true", "-printpriority=1"]))
+        connect_nodes(self.nodes[1], 0)
+        self.is_network_split=False
+        self.sync_all()
+
+    # Test before, at, and after expiry block
+    # Test case of reorgs
+    # Test mempool sync across nodes
+
+    def run_test(self):
+        alice = self.nodes[0].getnewaddress()
+        bob = self.nodes[1].getnewaddress()
+
+        # 11 kb blocks will only hold about 50 txs, so this will fill mempool with older txs
+        for _ in range(200):
+            self.nodes[0].sendtoaddress(bob, 0.5)
+        self.sync_all()
+
+        # Create some txs, send to other
+        tx1 = self.nodes[0].sendtoaddress(bob, 0.01)
+        self.sync_all()
+
+        # Test before, at, and after expiry block
+        print "block height:", self.nodes[0].getinfo()['blocks']
+        # Before expiry block height
+        assert(tx1 in self.nodes[0].getrawmempool())
+        # At expiry block height
+        self.nodes[0].generate(1)
+        self.sync_all()
+        rawtx = self.nodes[0].getrawtransaction(tx1, 1)
+        print "blockexpiry of tx1", rawtx['blockexpiry']
+        assert(tx1 in self.nodes[0].getrawmempool())
+        # After expiry block height, tx should leave mempool
+        blocks = self.nodes[0].generate(2)
+        for blk in blocks:
+            txs = self.nodes[0].getblock(blk)['tx']
+            if tx1 in txs:
+                print "Tx mined in block"
+            else:
+                print "Tx not in block"
```

Comment:
I can't tell whether this is expecting the transaction to be mined or dropped.

Reply:
This test has been rewritten